### PR TITLE
Elaborate on some adventure features

### PR DESCRIPTION
**Pull request by KingOfSquares**
books, bossbars, sounds, titles...

### CODE REVIEW

**Review by kashike — APPROVED** [review]

File: source/bossbar.rst [hunk 1]
```diff
@@ -6,16 +6,58 @@ Constructing a Boss Bar
...
-  final Component name = Component.text("Foo Bar");
-  final BossBar emptyBar = BossBar.bossBar(name, 0, BossBar.Color.RED, BossBar.Overlay.PROGRESS);
-  final BossBar halfBar = BossBar.bossBar(name, 0.5f, BossBar.Color.GREEN, BossBar.Overlay.NOTCHED_10);
-  final BossBar fullBar = BossBar.bossBar(name, 1, BossBar.Color.BLUE, BossBar.Overlay.NOTCHED_20);
-  // You can now send these Boss Bars to an audience
+  private @Nullable BossBar activeBar;
+
+  public void showMyBossBar(final @NonNull Audience target) {
+    final Component name = Component.text("Awesome BossBar");
+    //Creates a red boss bar which has no progress and no notches
```
Comment (zml2008) [inline source/bossbar.rst:23]:
```suggestion
    // Creates a red boss bar which has no progress and no notches
```

File: source/bossbar.rst [hunk 2]
```diff
@@ -6,16 +6,58 @@ Constructing a Boss Bar
...
+    final BossBar emptyBar = BossBar.bossBar(name, 0, BossBar.Color.RED, BossBar.Overlay.PROGRESS);
+    //Creates a green boss bar which has 50% progress and 10 notches
```
Comment (zml2008) [inline source/bossbar.rst:25]:
```suggestion
    // Creates a green boss bar which has 50% progress and 10 notches
```

File: source/bossbar.rst [hunk 3]
```diff
@@ -6,16 +6,58 @@ Constructing a Boss Bar
...
+    final BossBar halfBar = BossBar.bossBar(name, 0.5f, BossBar.Color.GREEN, BossBar.Overlay.NOTCHED_10);
+    //etc..
```
Comment (zml2008) [inline source/bossbar.rst:27]:
```suggestion
    // etc..
```

File: source/bossbar.rst [hunk 4]
```diff
@@ -6,16 +6,58 @@ Constructing a Boss Bar
...
+    final BossBar fullBar = BossBar.bossBar(name, 1, BossBar.Color.BLUE, BossBar.Overlay.NOTCHED_20);
+
+    //Send a bossbar to your audience
```
Comment (zml2008) [inline source/bossbar.rst:30]:
```suggestion
    // Send a bossbar to your audience
```

File: source/bossbar.rst [hunk 5]
```diff
@@ -6,16 +6,58 @@ Constructing a Boss Bar
...
+    target.showBossBar(fullBar);
+
+    //Store it locally to be able to hide it manually later
```
Comment (zml2008) [inline source/bossbar.rst:33]:
```suggestion
    // Store it locally to be able to hide it manually later
```

File: source/title.rst [hunk 6]
```diff
@@ -5,25 +5,32 @@ Titles
...
-^^^^^^^^^^^^^^^^
+.. code:: java
 
-By default, titles fade-in, stay on screen for a period of time and then fade-out.
+  public void showMyTitle(final @NonNull Audience target) {
+    final Component mainTitle = Component.text("This is the main title", NamedTextColor.WHITE);
+    final Component subtitle = Component.text("This is the subtitle", NamedTextColor.GRAY);
 
-.. code:: java
+    //Creates a simple title with the default values for fade-in, stay on screen and fade-out durations
```
Comment (zml2008) [inline source/title.rst:22]:
```suggestion
    // Creates a simple title with the default values for fade-in, stay on screen and fade-out durations
```

File: source/title.rst [hunk 7]
```diff
@@ -5,25 +5,32 @@ Titles
...
+    final Title title = Title.title(mainTitle, subtitle);
+
+    //Send the title to your audience
```
Comment (zml2008) [inline source/title.rst:25]:
```suggestion
    // Send the title to your audience
```

File: source/title.rst [hunk 8]
```diff
@@ -5,25 +5,32 @@ Titles
...
+    target.showTitle(title);
+  }
+
+  public void showMyTitleWithDurations(final @NonNull Audience target) {
+    final Title.Times times = Title.Times.of(Duration.ofMillis(500), Duration.ofMillis(3000), Duration.ofMillis(1000));
+    //Using the times object this title will use 500ms to fade in, stay on screen for 3000ms and then fade out for 1000ms
```
Comment (zml2008) [inline source/title.rst:31]:
```suggestion
    // Using the times object this title will use 500ms to fade in, stay on screen for 3000ms and then fade out for 1000ms
```

File: source/title.rst [hunk 9]
```diff
@@ -5,25 +5,32 @@ Titles
...
+    final Title title = Title.title(Component.text("Hello!"), Component.empty(), times);
 
-  // You can control the duration (in game ticks) of those animations by supplying a Times object
-  // In this example: The fade-in will take 15 ticks which correspond to 750ms
-  // After that, the title will remain for 3000ms and the finally fade-out within 20 ticks (1000ms)
-  final Title.Times times = Title.Times.of(Ticks.duration(15), Duration.ofMillis(3000), Ticks.duration(20));
-  final Title title2 = Title.title(Component.text("Hello!"), Component.empty(), times);
+    //Send the title, you can also use Audience#clearTitle() to remove the title at any time
```
Comment (zml2008) [inline source/title.rst:34]:
```suggestion
    // Send the title, you can also use Audience#clearTitle() to remove the title at any time
```

File: source/title.rst (hunk 8 above)
Comment (zml2008) [reply source/title.rst:31]: Also -- should mention these times are tick-based internally, so may not be precise if a client freezes

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Pull request merged**